build-logic/src/main/groovy/net/fabricmc/yarn/buildlogic/task/GeneratePackageInfoMappingsTask.groovy

```diff
@@ -0,0 +1,79 @@
+package net.fabricmc.yarn.buildlogic.task
+
+import groovy.transform.CompileStatic
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.tree.ClassNode
+
+import java.util.zip.ZipFile
+
+@CompileStatic
```

Comment:
Just out of curiosity, what does this annotation do?

Reply:
This annotation makes Groovy less dynamic, and makes it generate more Java-like bytecode (as far as I know)

Reply:
"This will let the Groovy compiler use compile time checks in the style of Java then perform static compilation" basically makes groovy a bit more type safe instead of it blowing up at runtime. Should really be put onto all groovy classes in this case.